liberty/lib/Planner/Planner.cpp

```diff
@@ -266,6 +279,69 @@ namespace liberty {
     assert(pdg != nullptr && "PDG is null?");
     writeGraph<PDG>(pdgName, pdg);
 
+    if(SlampCheck) {
+      uint32_t edgeCount = 0;
+      uint32_t diffCount = 0;
+      auto remed_slamp = &getAnalysis<SLAMPLoadProfile>(); 
+      auto remed_slamp_aa = std::make_unique<SlampOracleAA>(remed_slamp); 
+      auto remed_lamp = &getAnalysis<LAMPLoadProfile>(); 
+      auto remed_lamp_aa = std::make_unique<LampOracle>(remed_lamp); 
+
+      for(auto &edge : make_range(pdg->begin_edges(), pdg->end_edges())) {
+        if(!pdg->isInternal(edge->getIncomingT()) || !pdg->isInternal(edge->getOutgoingT()))
+          continue;
+
+        Instruction* src = dyn_cast<Instruction>(edge->getOutgoingT());
+        Instruction* dst = dyn_cast<Instruction>(edge->getIncomingT());
+        assert(src && dst && "src/dst not instructions in PDG?");
+        bool loopCarried = false;
+        edgeCount++;
+        DataDepType dataDepTy;
+        if(dyn_cast<CallBase>(src) || dyn_cast<CallBase>(dst))
+          continue;
+        if(edge->isRAWDependence())
+          dataDepTy = DataDepType::RAW;
+        else
+          continue;
+        if(edge->isLoopCarriedDependence())
+          loopCarried = true;
+        
+        auto slamp_remedy = remed_slamp_aa->memdep(src, dst, loopCarried, dataDepTy, loop);
+        auto lamp_remedy = remed_lamp_aa->memdep(src, dst, loopCarried, dataDepTy, loop);
+
+        if(slamp_remedy.depRes != lamp_remedy.depRes) {
+          diffCount++;
+          errs() << "SLAMP: " << slamp_remedy.depRes << ", LAMP: " << lamp_remedy.depRes << "\n";
+        }
+      }
+      errs() << "Total number of edges: " << edgeCount << ", number of diff edges: " << diffCount << "\n";
+    }
+
+    if(ValidityCheck) {
+      auto remed_slamp = &getAnalysis<SLAMPLoadProfile>(); 
+      auto remed_slamp_aa = std::make_unique<SlampOracleAA>(remed_slamp); 
+
+      for(auto nodeI : make_range(pdg->begin_nodes(), pdg->end_nodes())) {
+        for(auto nodeJ : make_range(pdg->begin_nodes(), pdg->end_nodes())) {
+           Instruction* src = dyn_cast<Instruction>(nodeI->getT());
+           Instruction* dst = dyn_cast<Instruction>(nodeJ->getT());
+
+           auto deps = pdg->getDependences(src, dst);
+           bool relevantDepExists = false;
+           for(auto dep : deps) {
+             if(dep->isMemoryDependence() && dep->isRAWDependence())
+               relevantDepExists = true;
+           }
+           if(!relevantDepExists) {
+             auto slamp_dep = remed_slamp_aa->memdep(src, dst, false, DataDepType::RAW, loop);
```

Comment:
the dependence also has `isLoopCarried`. Check those individually before querying. This will lead to missing cases (e.g., no loop carried dependence in PDG but slamp says there is